pkg/startup/startup.go

```diff
@@ -250,6 +250,21 @@ func configureAndCheckIPAddressSubnets(ctx context.Context, cli client.Interface
 		terminate()
 	}
 
+	if node.Spec.BGP.IPv4Address == "" && node.Spec.BGP.IPv6Address == "" {
+		if os.Getenv("CALICO_NETWORKING_BACKEND") != "none" {
+			log.Error("No IPv4 or IPv6 addresses configured or detected, required for Calico networking")
+			// Unrecoverable error, terminate to restart.
+			terminate()
+		} else {
+			log.Warn("No IPv4 or IPv6 addresses configured, and autodetection is not enabled. Some features may not work properly.")
```

Comment:
this log still refers to "autodetction", I think it should mirror the one above (configured or detected.)